### PR TITLE
Align item text in status menu horizontally

### PR DESCRIPTION
**Pull request by graphemecluster**
Align the start and the end of the labels.

### CODE REVIEW

File: CantoboardFramework/Keyboard/View/Status/StatusMenu.swift [hunk 1]
```diff
@@ -83,15 +80,18 @@ class StatusMenu: UIView {
         super.layoutSubviews()
         layer.shadowPath = CGPath(roundedRect: bounds, cornerWidth: Self.cornerRadius, cornerHeight: Self.cornerRadius, transform: nil)
         
+        let rowHeight = bounds.height / CGFloat(itemLabelInRows.count)
+        let font = UIFont.systemFont(ofSize: bounds.width * CGFloat(itemLabelInRows.count) * Self.fontSizeRatio) // intentional
```
Comment (alex-the-man) [inline CantoboardFramework/Keyboard/View/Status/StatusMenu.swift:84]: If I'm reading the code correctly, are you changing the font size calculation from height based to width based?

Comment (graphemecluster) [reply CantoboardFramework/Keyboard/View/Status/StatusMenu.swift]: Ah, OK. Now it should be mathematically the same while improving readability.

**Comment by graphemecluster** [comment]
Committed accordingly.

**Comment by alex-the-man** [comment]
Could you please describe your change with words? It's very difficult to guess your intention from the code. I would really appreciate if you could explain why and what the change is, for each PR.

Also, when I was coding this I was in a rush. If we want to do things properly, we should calculate the font size from the row height:

static let fontSizePerHeight: CGFloat = 20 / "＠".size(withFont: UIFont.systemFont(ofSize: 20)).height

let font = UIFont.systemFont(ofSize: bounds.width / CGFloat(itemLabelInRows[0].count) * Self.fontSizeRatio)

Note: The constant 20 is the initial guess of the font size. We should keep it close to the actual font size being used. As the font size and the font height in pixel isn't strictly linear, the calculated fontSizePerHeight will not be accurate enough.

**Comment by graphemecluster** [comment]
Replacing "font size per height" to "font size per width" does nothing at all.
The aim of this PR is to align the start and the end of the labels such that it looks pleasant to the user:
Screenshot

In other words, the cell width must be used to achieve this effect. Unit dimension does not do the trick.
Please think of a better method if you could.

Moreover, you've already set `adjustsFontSizeToFitWidth` to true, there's no need to worry about the content being too big.

File: CantoboardFramework/Keyboard/View/Status/StatusMenu.swift [hunk 2]
```diff
@@ -83,15 +80,18 @@ class StatusMenu: UIView {
         super.layoutSubviews()
         layer.shadowPath = CGPath(roundedRect: bounds, cornerWidth: Self.cornerRadius, cornerHeight: Self.cornerRadius, transform: nil)
         
+        let rowHeight = bounds.height / CGFloat(itemLabelInRows.count)
+        let font = UIFont.systemFont(ofSize: bounds.width / CGFloat(itemLabelInRows[0].count) * Self.fontSizeRatio)
```
Comment (alex-the-man) [inline CantoboardFramework/Keyboard/View/Status/StatusMenu.swift:84]: What does `bounds.width / CGFloat(itemLabelInRows[0].count)` mean?
How is Self.fontSizeRatio calculated?

**Comment by graphemecluster** [comment]
You're right, I found that the expected value of `fontSizeRatio` is `fontSizePerWidth / 8`, so I've updated accordingly.
`bounds.width / CGFloat(itemLabelInRows[0].count)` is the width of two cells, or a large cell spanning two columns.
`CGFloat(itemLabelInRows[0].count)` is 1 or 2 depending on `menuRows`.

**Comment by alex-the-man** [comment]
仲唔係好明點解嗰條算式。請問你嘅意圖係唔係想所有label都用同一字形，令到佢哋可以對齊？而條算式係你諗出嚟定係試出嚟嘅結果？

**Comment by alex-the-man** [comment]
How the constant 8 in `Self.fontSizePerWidth / 8` is derived:

telegram-cloud-photo-size-5-6136538973925912881-y

**Pull request merged**